internal/resource/config/session/authenticationsessionpolicies/global/session_authentication_session_policies_global_data_source.go

```diff
@@ -0,0 +1,137 @@
+				Computed:    true,
+				Optional:    false,
+			},
+			"hash_unique_user_key_attribute": schema.BoolAttribute{
+				Description: "Determines whether to hash the value of the unique user key attribute.",
+				Computed:    true,
+				Optional:    false,
+			},
+			"idle_timeout_mins": schema.Int64Attribute{
+				Description: "The idle timeout period, in minutes. If set to -1, the idle timeout will be set to the maximum timeout. The default is 60.",
+				Computed:    true,
+				Optional:    false,
+			},
+			"idle_timeout_display_unit": schema.StringAttribute{
+				Description: "The display unit for the idle timeout period in the PingFederate administrative console. When the display unit is HOURS or DAYS, the timeout value in minutes must correspond to a whole number value for the specified unit. [ MINUTES, HOURS, DAYS ]",
+				Computed:    true,
+				Optional:    false,
+			},
+			"max_timeout_mins": schema.Int64Attribute{
+				Description: "The maximum timeout period, in minutes. If set to -1, sessions do not expire. The default is 480.",
+				Computed:    true,
+				Optional:    false,
+			},
+			"max_timeout_display_unit": schema.StringAttribute{
+				Description: "The display unit for the maximum timeout period in the PingFederate administrative console. When the display unit is HOURS or DAYS, the timeout value in minutes must correspond to a whole number value for the specified unit.",
+				Computed:    true,
+				Optional:    false,
+			},
+		},
+	}
+
+	id.ToSchema(&schema)
+	resp.Schema = schema
+}
+
+// Metadata returns the resource type name.
+func (r *sessionAuthenticationSessionPoliciesGlobalDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_session_authentication_session_policies_global"
+}
+
+func (r *sessionAuthenticationSessionPoliciesGlobalDataSource) Configure(_ context.Context, req datasource.ConfigureRequest, _ *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	providerCfg := req.ProviderData.(internaltypes.ResourceConfiguration)
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+
+}
+
+func readSessionAuthenticationSessionPoliciesGlobalDataSource(ctx context.Context, r *client.GlobalAuthenticationSessionPolicy, state *sessionAuthenticationSessionPoliciesGlobalDataSourceModel) {
+	state.Id = types.StringValue("session_authentication_session_policies_global_id")
+	state.EnableSessions = types.BoolValue(r.EnableSessions)
+	state.PersistentSessions = types.BoolPointerValue(r.PersistentSessions)
+	state.HashUniqueUserKeyAttribute = types.BoolPointerValue(r.HashUniqueUserKeyAttribute)
+	state.IdleTimeoutMins = types.Int64PointerValue(r.IdleTimeoutMins)
+	state.IdleTimeoutDisplayUnit = internaltypes.StringTypeOrNil(r.IdleTimeoutDisplayUnit, true)
```

Comment:
Do we need this method anymore? Can we just use types.StringPointerValue?

Reply:
I think in a data source it makes sense to go with the framework default.  Changes made.